test/fuzz/cse.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Halide.h"
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include <time.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+using std::vector;
+
+Expr random_expr(FuzzedDataProvider &fdp, int depth, vector<Expr> &exprs) {
+    if (depth <= 0) {
+        return fdp.ConsumeIntegralInRange<int>(-5, 4);
+    }
+
+    if (!exprs.empty() && fdp.ConsumeBool()) {
+        // Reuse an existing expression
+        return exprs[fdp.ConsumeIntegralInRange<size_t>(0, exprs.size() - 1)];
+    }
+
+    Expr next;
+    switch (fdp.ConsumeIntegralInRange<int>(0, 8)) {
```

Comment:
This looks ok.

Reply:
If the idea is that a value of 8 is intended to hit the default case, we should add a comment to that effect

Reply:
yeah that's what it looks like, I was mostly just attempting to replicate the original logic but while I'm at it I can probably make this clearer too.